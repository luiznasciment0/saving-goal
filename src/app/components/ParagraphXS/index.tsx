import Typography, { WeightProps, ColorProps } from 'app/components/Typography'

interface ParagraphXSProps {
  children: React.ReactNode
  weight?: WeightProps
  color?: ColorProps
}

const ParagraphXS = ({
  children,
  weight = 'light',
  color = 'primaryGray'
}: ParagraphXSProps) => (
  <Typography as="h3" weight={weight} size="small" color={color}>
    {children}
  </Typography>
)

export default ParagraphXS
