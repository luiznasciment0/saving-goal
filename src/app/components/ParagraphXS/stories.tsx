import { Story, Meta } from '@storybook/react/types-6-0'
import ParagraphXS from '.'

export default {
  title: 'ParagraphXS',
  component: ParagraphXS,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => (
  <ParagraphXS>{args.children}</ParagraphXS>
)

Default.args = {
  children: 'Test'
}
