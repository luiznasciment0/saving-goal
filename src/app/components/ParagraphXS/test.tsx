import { render, screen } from '@testing-library/react'

import ParagraphXS from '.'

describe('<ParagraphXS />', () => {
  it('should render the ParagraphXS and its styles', () => {
    const { container } = render(<ParagraphXS>Test</ParagraphXS>)

    expect(screen.getByRole('heading', { name: /test/i })).toHaveStyle({
      'font-size': '1.4rem',
      'line-height': '2.1rem',
      'font-weight': '400',
      'font-family': `'Work Sans'`,
      color: '#708797'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
