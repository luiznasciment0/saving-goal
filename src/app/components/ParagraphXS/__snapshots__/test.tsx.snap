// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParagraphXS /> should render the ParagraphXS and its styles 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 2.1rem;
  font-family: 'Work Sans';
  color: #708797;
  font-weight: 400;
}

<h3
  class="c0"
  color="primaryGray"
>
  Teste
</h3>
`
